synapse/rest/client/room.py

```diff
@@ -1166,7 +1166,7 @@ async def on_GET(
             )
 
         limit = parse_integer(request, "limit")
-        if limit is not None and limit <= 0:
```

Comment:
I understand "positive" to mean "> 0".

If passing `limit=0` constitutes a valid request, I'd suggest the error message should say "non-negative" (i.e. >=0) instead of "positive".

The terminology is a little awkward, but standard AFAIK.

Reply:
Agreed. MSC says

> * **`limit`**: Optional: a client-defined limit to the maximum number of rooms to return per page. Must an integer greater than zero.

So it was correct before.

Reply:
I'll back out that commit, must have confused myself at some point!